dem_sim/training.py

```diff
@@ -26,17 +27,28 @@ def train(
         total_loss = 0
         for i, (graph_data, step) in enumerate(loader):
             graph_data, step = _unbatch(graph_data, step)
+            graph_data = graph_data.copy_to(device)
+
             prediction = simulator(graph_data, step)
             loss = loss_function(prediction, graph_data, step)
             metric.add(prediction, graph_data, step)
             loss.backward()
             total_loss += loss.item()
             optimizer.step()
+            wandb.log({"loss": loss})
+            
         total_loss = total_loss / (i + 1)
         losses.append(total_loss)
+
         print(f"Loss after epoch {epoch}: {total_loss:.4E}...")
         print(metric)
 
+        wandb.log({"loss_epoch": total_loss})
+        wandb.log({"rmse_positions": metric.positions})
```

Comment:
Here also we can avoid listing all the different metrics by adding something like below to the metric class and here doing just `wandb.log(metric.dict)`

```
    @property
    def dict(self):
        return {"rms_" + key: getattr(self, key) for key in self.attributes}
```

Reply:
Added functionality, slightly different in commit [467a7b9](https://github.com/GrainLearning/graphs/pull/3/commits/467a7b9a1bf7433321f1f3e10d6be87c19c9ecc1)